easybuild/easyblocks/generic/systemcompiler.py

```diff
@@ -38,6 +38,29 @@
 from easybuild.tools.build_log import EasyBuildError
 
 
+def extract_compiler_version(compiler_name):
+    """Extract compiler version for provided compiler_name."""
+    # look for 3-4 digit version number, surrounded by spaces
+    # examples:
+    # gcc (GCC) 4.4.7 20120313 (Red Hat 4.4.7-11)
+    # Intel(R) C Intel(R) 64 Compiler XE for applications running on Intel(R) 64, Version 15.0.1.133 Build 20141023
+    if compiler_name == 'gcc':
+        out, _ = run_cmd("gcc --version", simple=False)
+    elif compiler_name in ['icc', 'ifort']:
+        out, _ = run_cmd("%s -V" % compiler_name, simple=False)
+    else:
+        raise EasyBuildError("Unknown compiler %s" % compiler_name)
+
+    version_regex = re.compile(r'\s([0-9]+(?:\.[0-9]+){1,3})\s', re.M)
+    res = version_regex.search(out)
+    if res:
+        compiler_version = res.group(1)
+        debug_msg = "Extracted compiler version '%s' for %s from: %s", compiler_version, compiler_name, out
+        return compiler_version, debug_msg
```

Comment:
don't pass down the debug log message, just log here; also, we try to avoid 'inline' `return` statements

```python
_log = fancylogger.getLogger('easyblocks.generic.systemcompiler')
...
def extract_compiler_version(compiler_name):

    ...
    if res:
        compiler_version = res.group(1)
        _log.debug("Extracted compiler version '%s' for %s from: %s", compiler_version, compiler_name, out)
    else:
        raise EasyBuildError("...")

    return compiler_version
```